src/ExceptionTypes.f90

```diff
@@ -0,0 +1,370 @@
+
+      counterOut = this%counter
+
+    ENDFUNCTION getCounter_ExceptionTypeBase
+!
+!-------------------------------------------------------------------------------
+!> @brief Logs mesg to file
+!>
+    SUBROUTINE logMessage_ExceptionTypeBase(this,isQuiet,isLogActive,logUnit,mesg)
+      CLASS(ExceptionTypeBase),INTENT(INOUT) :: this
+      LOGICAL(SBK),INTENT(IN) :: isQuiet
+      LOGICAL(SBK),INTENT(IN) :: isLogActive
+      INTEGER(SIK),INTENT(IN) :: logUnit
+      CHARACTER(LEN=EXCEPTION_MAX_MESG_LEN),INTENT(INOUT) :: mesg
+      CHARACTER(LEN=EXCEPTION_MAX_MESG_LEN) :: prefix
+      INTEGER(SIK) :: ioerr1,ioerr2,prefixLen
+
+      REQUIRE(this%isInit)
+      CALL this%genPrefix(prefix)
+
+      !Write to the default standard error output
+      IF(.NOT.isQuiet) THEN
+        WRITE(ERROR_UNIT,'(a)') TRIM(prefix)
+        WRITE(ERROR_UNIT,'(6x,a)') TRIM(mesg)
+        FLUSH(ERROR_UNIT)
+      ENDIF
+
+      !Write to the log file
+      IF(isLogActive) THEN
+        WRITE(logUnit,'(a)',IOSTAT=ioerr1) TRIM(prefix)
+        WRITE(logUnit,'(6x,a)',IOSTAT=ioerr2) TRIM(mesg)
+        FLUSH(logUnit)
+
+        !Additional error message if problem writing to log file
+        IF(ioerr1 /= 0 .OR. ioerr2 /= 0) THEN
+          WRITE(ERROR_UNIT,'(a)') '#### EXCEPTION_INFORMATION ####'
+          IF(isQuiet) THEN
+            !If quiet mode, then print the message that failed.
+            WRITE(ERROR_UNIT,'(6x,a)') 'Problem writing to log file.'
+            WRITE(ERROR_UNIT,'(6x,a)') 'Original Message: "'// &
+                                       TRIM(prefix)//' - '//TRIM(mesg)//'"'
+          ELSE
+            !Message was already printed.
+            WRITE(ERROR_UNIT,'(6x,a)') 'Problem writing above message '// &
+                                       'to log file.'
+          ENDIF
+        ENDIF
+      ENDIF
+
+      !Set the message to be included as one line back to exception object
+      prefixLen = LEN_TRIM(prefix)+3
+      WRITE(mesg,'(a)') TRIM(prefix)//' - '//TRIM(mesg(1:EXCEPTION_MAX_MESG_LEN-prefixLen))
+    ENDSUBROUTINE logMessage_ExceptionTypeBase
+!
+!-------------------------------------------------------------------------------
+!> @brief Get tag
+!>
+    PURE FUNCTION getTag_ExceptionTypeBase(this) RESULT(tag)
+      CLASS(ExceptionTypeBase),INTENT(IN) :: this
+      INTEGER(SIK) :: tag
+
+      tag = this%tag
+
+    ENDFUNCTION getTag_ExceptionTypeBase
+!
+!-------------------------------------------------------------------------------
+!> @brief Stops execution based on value of stopmode
+!> @param stopmode boolean to indicate whether or not to stop execution
+!>
+    SUBROUTINE exceptionStop(stopmode)
+#ifdef HAVE_MPI
+      INCLUDE 'mpif.h'
+      INTEGER(SIK) :: ierr
+#endif
+      LOGICAL(SBK),INTENT(IN) :: stopmode
+#ifdef HAVE_MPI
+      IF(stopmode) CALL MPI_Abort(MPI_COMM_WORLD,666,ierr)
+#else
+      IF(stopmode) THEN
+        STOP 666
+      ENDIF
+#endif
+    ENDSUBROUTINE exceptionStop
```

Comment:
If this needs to exist, it probably only makes sense in the ExceptionHandlerType module since it isn't specific to a particular type of exception